scripts/odb_extractor_atoms.py

```diff
@@ -154,104 +538,174 @@ def find_atoms(observation):
-    """
+            print(f"{obsnum[ii], obsid}", file=fid)
+            obs_class = group[item]['obsClass'].upper()
+            phase2stat = group[item]['phase2Status'].upper()
+            obs_stat = group[item]['obsStatus'].upper()
+            #             print(obs_class, phase2stat, obs_stat)
+            if obs_class in sel_obs_class and obs_stat in sel_obs_status:
+                if wb is not None:
+                    ws = wb.create_sheet(title=obsid.split('-')[-1])
+                #                     ws['A1'] = obsid
+
+                # Atoms in each sequence
+                atoms = findatoms(group[item], verbose=verbose, ws=ws, fid=fid)
+                # Summary of atoms
+                #                 classes = []
+                #                 qastates = []
+                #                 for atom in atoms:
+                #                     print('Atom ', atom['id'])
+                #                     for key in atom.keys():
+                #                         print(f" \t {key}: {atom[key]}")
+                #                         if key == 'class':
+                #                             classes.append(atom[key])
+                #                         if key == 'qa_state':
+                #                             qastates.append(atom[key])
+                #                             if atom[key].upper() == 'PASS':
+                #                                 obs_program_used += atom['prog_time']
+                #                                 obs_partner_used += atom['part_time']
+                #                 obsclass = select_obsclass(classes)
+                #                 print(f" Obsclass: {obsclass}")
+                #                 obs_qastate = select_qastate(qastates)
+                #                 print(f" QAstate (atoms): {obs_qastate}")
+                #                 print(f" qaState (ODB): {group[item]['qaState']}")
+                #                 if group[item]['qaState'].upper() == 'PASS':
+                #                     if group[item]['obsClass'] in ['science', 'progCal']:
+                #                         obs_program_used += float(group[item]['setupTime'])/1000.
+                #                     elif group[item]['obsClass'] in ['partnerCal']:
+                #                         obs_partner_used += float(group[item]['setupTime'])/1000.
+
+                #                 print(f" program_used: {obs_program_used}")
+                #                 print(f" partner_used: {obs_partner_used}")
+                print('', file=fid)
+
+            print('', file=fid)
 
+    return
+
+
+def prog_proc(program, sel_obs_class=['SCIENCE', 'PROGCAL', 'PARTNERCAL', 'ACQ', 'ACQCAL', 'DAYCAL'], \
+              sel_obs_status=['PHASE_2', 'FOR_REVIEW', 'IN_REVIEW', 'FOR_ACTIVATION', 'ON_HOLD', 'READY', \
+                              'ONGOING', 'OBSERVED', 'INACTIVE'], fid=sys.stdout, xls=None, verbose=False):
```

Comment:
Again, this string repetition is hardly ideal and should be conglomerated to `Enum`s if they don't already exist (and I think they do).